ext/liquid_c/vm.c

```diff
@@ -273,9 +273,10 @@ static VALUE vm_render_until_error(VALUE uncast_args)
                 break;
             case OP_RENDER_VARIABLE_RESCUE:
                 // Save state used by vm_render_rescue to rescue from a variable rendering exception
-                args->node_line_number = (unsigned int)*const_ptr++;
+                args->node_line_number = (ip[0] << 16) | (ip[1] << 8) | ip[2];
```

Comment:
We don't really need to decode the number immediately.  The number is only needed when rescuing from an exception.  We could instead store a pointer to the line number bytes here, then decode them in `vm_render_rescue`